prophet/src/error.rs

```diff
@@ -0,0 +1,25 @@
+/// This type represents all possible errors that can occur when analyzing
+/// a microservice project
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum Error {
```

Comment:
The name of this struct seems a little generic. Could we call this `AnalysisError`, `ProphetAnalysisError`, or something like that?

Reply:
It's standard practice in Rust to use module paths instead for this kind of thing rather than adding redundant names to the types themselves (see [here](https://docs.serde.rs/serde/?search=Error) for example), so this could be referenced as `prophet::Error` rather than `prophet::ProphetError`

Reply:
Ah, I see. Thanks for letting me know!